tests/util_test.py

```diff
@@ -63,6 +64,21 @@ def kw_to_positional(factor, *args, **kwargs):
     self.assertEqual(dict(three=6, four=8), scaled_kwargs)
     self.assertEqual(2, out_thunk())
 
+  def test_weakref_lru_cache(self):
```

Comment:
Add a threaded test that exhibits the bug?

Reply:
I could not get the bug to arise in any reasonable amount of time without carefully delaying specific lines of code. (a sleep(0) between iter and next is necessary to trigger the bug). The bug is also unrelated to threading. This test, however, does deadlock if you put a lock in the weakref callback.

Reply:
It's unrelated to threading? Wasn't the original report only for threaded code, and presumably the easy way to make in between those two statements is to trigger GC of the weakref from another thread?

Reply:
I tried testing with threads and it didn't help reproduce the problem any more. The problem comes when GC is triggered which does not change based on the number of threads. All the other code is threadsafe where only one python thread can be in the critical section at a time, so more threads do not improve the chances of a problem happening. I suspect that the multiple threads in the user code just helped trigger this code because it was hitting the cache more.